src/tabnineChatWidget/ChatApi.ts

```diff
@@ -94,162 +90,167 @@ type WorkspaceFolders = {
+          throw new Error("state has no access token");
+        }
+        return {
+          token: state.access_token,
+          username: state.user_name,
+          avatarUrl: state.user_avatar_url,
+          serviceLevel: state.service_level,
+        };
+      })
+      .registerEvent<void, GetCapabilitiesResponse>(
+        "get_capabilities",
+        async () => {
+          const capabilitiesResponse = await getCapabilities();
+          if (!capabilitiesResponse) {
+            throw new Error("capabilities response is undefined");
+          }
+          return {
+            enabledFeatures: capabilitiesResponse.enabled_features,
+          };
+        }
+      )
+      .registerEvent<SendEventRequest, void>(
+        "send_event",
+        async (req: SendEventRequest) => {
+          await sendEvent({
+            name: req.eventName,
+            properties: req.properties,
+          });
+        }
+      )
+      .registerEvent<void, BasicContext>("get_basic_context", getBasicContext)
+      .registerEvent<
+        EnrichingContextRequestPayload,
+        EnrichingContextResponsePayload
+      >("get_enriching_context", getEnrichingContext)
+      .registerEvent<void, SelectedCodeResponsePayload>(
+        "get_selected_code",
+        getSelectedCode
+      )
+      .registerEvent<InserCode, void>("insert_at_cursor", insertTextAtCursor)
+      .registerEvent<
+        { symbol: string },
+        vscode.SymbolInformation[] | undefined
+      >("resolve_symbols", resolveSymbols)
+      .registerEvent<{ symbols: vscode.SymbolInformation[] }, void>(
+        "peek_definition",
+        peekDefinition
+      )
+      .registerEvent<NavigateToLocationPayload, void>(
+        "navigate_to_location",
+        navigateToLocation
+      )
+      .registerEvent<ChatConversation, void>(
+        "update_chat_conversation",
+        async (conversation) => {
+          const chatState = context.globalState.get(CHAT_CONVERSATIONS_KEY, {
+            conversations: {},
+          }) as ChatState;
+          chatState.conversations[conversation.id] = {
+            id: conversation.id,
+            messages: conversation.messages,
+          };
+          await context.globalState.update(CHAT_CONVERSATIONS_KEY, chatState);
+        }
+      )
+      .registerEvent<void, ChatState>(
+        "get_chat_state",
         () =>
           context.globalState.get(CHAT_CONVERSATIONS_KEY, {
             conversations: {},
           }) as ChatState
       )
-    );
-  }
-
-  chatEventRegistry
-    .registerEvent<void, InitResponse>("init", async () => {
-      onInit();
-      return Promise.resolve({
-        ide: "vscode",
-        isDarkTheme: [
-          ColorThemeKind.HighContrast,
-          ColorThemeKind.Dark,
-        ].includes(vscode.window.activeColorTheme.kind),
-        isTelemetryEnabled: isCapabilityEnabled(Capability.ALPHA_CAPABILITY),
-        serverUrl,
-        isSelfHosted: context.workspaceState.get(
```

Comment:
@yonidavidson this was changed to be injected from the ctor, and not managed as a flag

Reply:
meant @yonip23  and not @yonidavidson  🤦‍♂️